src/WebJobs.Script.WebHost/Diagnostics/SystemLogger.cs

```diff
@@ -173,9 +176,28 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
                     functionName = string.IsNullOrEmpty(fex.MethodName) ? string.Empty : fex.MethodName.Replace("Host.Functions.", string.Empty);
                 }
 
-                (innerExceptionType, innerExceptionMessage, details) = exception.GetExceptionDetails();
-                formattedMessage = Sanitizer.Sanitize(formattedMessage);
-                innerExceptionMessage = innerExceptionMessage ?? string.Empty;
+                var baseEx = exception.GetBaseException();
+                innerExceptionType = baseEx.GetType().FullName;
+
+                var originalMessage = baseEx.Message;
```

Comment:
All these code changes are making it hard to see if you're regressing anything here. Can't you just keep the existing codepath as is and add if/else handling for these invocation exceptions?

E.g. currently the formatted message is being sanitized via Sanitizer.Sanitize(formattedMessage). In your refactoring you've lost that. Not sure if that is intentional.

Reply:
This is intentional - formatted messages won't contain any secrets unless we explicitly choose to include them in the logs. I removed it for efficiency, but I can restore it if you believe there's any potential risk.